lib/streamlit/elements/number_input.py

```diff
@@ -77,11 +77,49 @@ def number_input(
         >>> st.write('The current number is ', number)
         """
 
+        # Ensure that all arguments are of the same type.
+        args = [min_value, max_value, value, step]
+
+        int_args = all(
```

Comment:
I know this is copied from below, but now that the list of types we're checking against is growing, could we shorten this to the following?
```python
int_args = all(isinstance(a, (numbers.Integral, type(None), NoValue)) for a in args)
float_args = all(isinstance(a, (float, type(None), NoValue)) for a in args)
```

Reply:
I concur. I just pushed a commit with this change.